### PR TITLE
Remove old, core/v1 specific constructs from RESTClient

### PR DESCRIPTION
**Pull request by smarterclayton**
Now that metav1 is abstracted from the APIs, RESTClient should also be agnostic to the core API.

* Remove `LabelSelectorParam` and `FieldSelectorParam` - use `VersionedParams` with `ListOptions`
* Remove `UintParam`
* Remove all legacy field selector logic from `VersionedParams` - ParameterCodec now handles that
* Remove special parameters (like `timeout`) which is no longer set by most clients

```release-note
The generic RESTClient type (`k8s.io/client-go/rest`) no longer exposes `LabelSelectorParam` or `FieldSelectorParam` methods - use `VersionedParams` with `metav1.ListOptions` instead.  The `UintParam` method has been removed.  The `timeout` parameter will no longer cause an error when using `Param()`.
```

### CODE REVIEW

**Comment by smarterclayton** [comment]
@kubernetes/sig-api-machinery-pr-reviews

**Comment by smarterclayton** [comment]
This makes resource.Helper a versioned interface (no field mapping) which it already was for non core API resources.

File: pkg/kubectl/resource/helper.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ func (m *Helper) Get(namespace, name string, export bool) (runtime.Object, error
 		Resource(m.Resource).
 		Name(name)
 	if export {
+		// TODO: I should be part of GetOptions
```
Comment (wojtek-t) [inline pkg/kubectl/resource/helper.go]: s/I/It/

**Review by wojtek-t — COMMENTED** [review]
Nice - just one very minor comment.

**Pull request merged**